aisdc/attacks/attack_report_formatter.py

```diff
@@ -28,14 +28,31 @@ def __init__(self, filename=None):
         os.makedirs(dirname, exist_ok=True)
         # if file doesn't exist, create it
         if not os.path.exists(self.filename):
-            with open(self.filename, "w+", encoding="utf-8") as f:
+            with open(self.filename, "w", encoding="utf-8") as f:
                 f.write("")
 
-    def add_attack_output(self, incoming_json):
+    def add_attack_output(self, incoming_json, class_name):
         """Add a section of JSON to the file which is already open"""
-        with open(self.filename, "a", encoding="utf-8") as f:
-            f.write(incoming_json)
-            f.write("\n")
+
+        # Read the contents of the file and then clear the file
+        with open(self.filename, "r+", encoding="utf-8") as f:
+            file_contents = f.read()
+            if file_contents != "":
+                file_data = json.loads(file_contents)
+            else:
+                file_data = {}
+
+            f.truncate(0)
+
+        # Add the new JSON to the JSON that was in the file, and re-write
+        with open(self.filename, "w", encoding="utf-8") as f:
+            incoming_json = json.loads(incoming_json)
+
+            if "log_id" in incoming_json:
```

Comment:
@yolaj-nhs 

Thanks, overall this PR seems great to me. 

I wonder how it would be possible to have a log_id repeated. I believe the way we implemented it, each time the make_report(...) method is called, a unique UUID is supposed to be generated. Though I agree that storing output results from multiple attacks with the top-level JSON component as AttackName_LogID would be useful.

Reply:
@yolaj-nhs don't think collisions are likely to be an issue. More pressing things to do :)